src/text/text_formatter.rs

```diff
@@ -0,0 +1,401 @@
+        //                   ^-- delimiting colon, formatted second
+        value.fmt_fractional_seconds(&mut *self.output)?;
+
+        self.format_offset(offset_minutes)?;
+        Ok(())
+    }
+
+    fn format_offset(&mut self, offset_minutes: Option<i32>) -> IonResult<()> {
+        if offset_minutes.is_none() {
+            write!(self.output, "-00:00")?;
+            return Ok(());
+        }
+        let offset_minutes = offset_minutes.unwrap();
+
+        const MINUTES_PER_HOUR: i32 = 60;
+        // Split the offset into a sign and magnitude for formatting
+        let sign = if offset_minutes >= 0 { "+" } else { "-" };
+        let offset_minutes = offset_minutes.abs();
+        let hours = offset_minutes / MINUTES_PER_HOUR;
+        let minutes = offset_minutes % MINUTES_PER_HOUR;
+        write!(self.output, "{}{:0>2}:{:0>2}", sign, hours, minutes)?;
+        Ok(())
+    }
+
+    pub(crate) fn format_symbol<A: AsRawSymbolTokenRef>(&mut self, value: A) -> IonResult<()> {
+        RawTextWriter::write_symbol_token(&mut BufWriter::new(&mut *self.output), value)?;
+        Ok(())
+    }
+
+    pub(crate) fn format_string<S: AsRef<str>>(&mut self, value: S) -> IonResult<()> {
+        write!(self.output, "\"")?;
+        RawTextWriter::write_escaped_text_body(&mut BufWriter::new(&mut *self.output), value)?;
+        write!(self.output, "\"")?;
+        Ok(())
+    }
+
+    pub(crate) fn format_clob<A: AsRef<[u8]>>(&mut self, value: A) -> IonResult<()> {
+        // clob_value to be written based on defined STRING_ESCAPE_CODES.
+        const NUM_DELIMITER_BYTES: usize = 4; // {{}}
+        const NUM_HEX_BYTES_PER_BYTE: usize = 4; // \xHH
+
+        let value: &[u8] = value.as_ref();
+
+        // Set aside enough memory to hold a clob containing all hex-encoded bytes
+        let mut clob_value =
+            String::with_capacity((value.len() * NUM_HEX_BYTES_PER_BYTE) + NUM_DELIMITER_BYTES);
+
+        for byte in value.iter().copied() {
+            let c = byte as char;
+            let escaped_byte = &self.string_escape_codes[c as usize];
+            if !escaped_byte.is_empty() {
+                clob_value.push_str(escaped_byte);
+            } else {
+                clob_value.push(c);
+            }
+        }
+        write!(self.output, "{{{{\"{}\"}}}}", clob_value)?;
+        Ok(())
+    }
+
+    pub(crate) fn format_blob<A: AsRef<[u8]>>(&mut self, value: A) -> IonResult<()> {
+        write!(self.output, "{{{{{}}}}}", base64::encode(value))?;
+        Ok(())
+    }
+
+    pub(crate) fn format_struct(&mut self, value: &OwnedStruct) -> IonResult<()> {
+        write!(self.output, "{{ ")?;
+        let mut peekable_itr = value.iter().peekable();
+        while peekable_itr.peek() != None {
+            let (field_name, field_value) = peekable_itr.next().unwrap();
+            self.format_symbol(field_name.text().unwrap())?;
+            write!(self.output, ": ")?;
+            write!(self.output, "{}", field_value)?;
```

Comment:
We can merge these to make:
```suggestion
            write!(self.output, ": {}", field_value)?;
```